src/tiled/tilesetdock.cpp

```diff
@@ -758,20 +758,33 @@ void TilesetDock::exportTileset()
     if (!tileset)
         return;
 
+    Preferences *prefs = Preferences::instance();
+
     const QLatin1String extension(".tsx");
-    QString suggestedFileName = QFileInfo(mMapDocument->fileName()).path();
-    suggestedFileName += QLatin1Char('/');
+    QLatin1String empty(".");
+    QString suggestedFileName = prefs->lastPath(Preferences::ExternalTileset);
+
+    for(int i = (suggestedFileName.length() - 1); suggestedFileName.at(i) != QLatin1Char('/')
+        && suggestedFileName != empty; i--)
+        suggestedFileName.remove(i, 1);
+
+    if(suggestedFileName == empty)
+        QString suggestedFileName = QFileInfo(mMapDocument->fileName()).path();
+        suggestedFileName += QLatin1Char('/');
```

Comment:
The `if` condition here seems broken. Why do you expect the `suggestedFileName` to be "."? I think you want to put this check above the "extracting of the path" part, and just use `suggestedFileName.isEmpty()`.